src/hope_payment_gateway/apps/gateway/flows.py

```diff
@@ -98,6 +104,8 @@ def _get_object_status(self) -> str:
     @state.on_success()
     def _on_transition_success(self, description: str, source: str, target: str) -> None:
         self.object.save()
+        manager = initialize_engine()
+        manager.notify("payment.record.update", make_event(f"Record Status Update: {target}", key=self.object.pk))
```

Comment:
This is tricky: key is for extra route info (es. CountryOffice) message_id can be used as object id. BTW are only guidelines.